clean all warning complaining about comparison between signed and unsigned

Exemple
```
src/counters_module.cpp:28:29: warning: comparison of integer expressions of different signedness: ‘int’ and ‘const UINT’ {aka ‘const unsigned int’} [-Wsign-compare]
   28 |           for (int i = 0; i < counters::size; ++i)
      |                           ~~^~~~~~~~~~~~~~~~
```

--- a/.gitignore
+++ b/.gitignore
@@ -341,3 +341,8 @@ healthchecksdb
 
 Pin
 out
+/code_generator/just_for_cmake.cpp
+/include/replace/wrappers/avx.h
+/include/replace/wrappers/avx512.h
+/include/replace/wrappers/fma.h
+/include/replace/wrappers/sse.h

--- a/Test/test_counters/CMakeLists.txt
+++ b/Test/test_counters/CMakeLists.txt
@@ -81,7 +81,7 @@ function(configure_file FILENAME)
       DEPENDS
         ${CMAKE_GENERICS_PATH}/GenericConfigureFile.cmake
         #${CMAKE_CURRENT_SOURCE_DIR}/${FILENAME}
-      COMMENT "Configuring FILENAME"
+      COMMENT "Configuring ${FILENAME}"
       VERBATIM)
 endfunction(configure_file)
 

--- a/Test/test_replace_avx/replace.cpp
+++ b/Test/test_replace_avx/replace.cpp
@@ -2,12 +2,12 @@
 #include <immintrin.h>
 #include <string>
 
-int main(int argc, char* argv[])
+int main(int argc, const char* argv[])
 {
     std::cout << "This program is used for tests purposes only. "
         << std::endl;
     if (argc == 1) {
-        argv = new char* [3]{ "main", "4", "6" };
+        argv = new const char* [3]{ "main", "4", "6" };
     }
     float a = std::stof(argv[1]);
     float b = std::stof(argv[2]);

--- a/code_generator/templates/wrapper_template.txt
+++ b/code_generator/templates/wrapper_template.txt
@@ -40,30 +40,30 @@ namespace pene
         
         {% if architecture_name == "avx512"  %}
         template<typename T, void (*backend_fct)(T, T, T*,void*)>
-        void call_backend_fct(T* const a, T* const b, T* c, void* ctx, int i)
+        void call_backend_fct(T* const a, T* const b, T* c, void* ctx, size_t i)
         {
             backend_fct(a[i], b[i], &c[i], ctx);
         }
         {% elif architecture_name == "sse" or architecture_name  == "avx" %}
         template<typename T, void (*backend_fct)(T, T, T*,void*)>
-        void call_backend_fct(T* const a, T* const b, T* c, void* ctx, int N)
+        void call_backend_fct(T* const a, T* const b, T* c, void* ctx, size_t N)
         {
-            for(int i=0 ;i<N; ++i)
+            for(size_t i=0 ;i<N; ++i)
                backend_fct(a[i], b[i], &c[i], ctx);
         }
         {% endif %}
 
         {% if architecture_name == "avx512" or architecture_name == "fma" %}
         template<typename T>
-        void PIN_FAST_ANALYSIS_CALL zero_reg_at_position(T* dst, int pos)
+        void PIN_FAST_ANALYSIS_CALL zero_reg_at_position(T* dst, size_t pos)
         {
-        dst[pos]=0;
+            dst[pos]=0;
         }
-        static bool PIN_FAST_ANALYSIS_CALL msk_element_set(IMULTI_ELEMENT_OPERAND* OpInfo, int pos)
+        static bool PIN_FAST_ANALYSIS_CALL msk_element_set(IMULTI_ELEMENT_OPERAND* OpInfo, size_t pos)
         {
           return OpInfo->ElementMaskValue(pos)==1;
         }
-        static bool PIN_FAST_ANALYSIS_CALL msk_element_notset(IMULTI_ELEMENT_OPERAND* OpInfo, int pos)
+        static bool PIN_FAST_ANALYSIS_CALL msk_element_notset(IMULTI_ELEMENT_OPERAND* OpInfo, size_t pos)
         {
           return OpInfo->ElementMaskValue(pos)==0;
         }
@@ -71,25 +71,25 @@ namespace pene
         
         {% if architecture_name == "fma" %}
         template<typename T, void (*backend_fct_fmadd)(T, T, T, T*,void*)>
-        void call_backend_fct_fmadd(T* const a, T* const b, T* const c, T* d, void* ctx, int i)
+        void call_backend_fct_fmadd(T* const a, T* const b, T* const c, T* d, void* ctx, size_t i)
         {
             backend_fct_fmadd(a[i], b[i], c[i], &d[i], ctx);
         }
 
         template<typename T, void (*backend_fct_fnmadd)(T, T, T, T*,void*)>
-        void call_backend_fct_fnmadd(T* const a, T* const b, T* const c, T* d, void* ctx, int i)
+        void call_backend_fct_fnmadd(T* const a, T* const b, T* const c, T* d, void* ctx, size_t i)
         {
             backend_fct_fnmadd(-a[i], b[i], c[i], &d[i], ctx);
         }
         
          template<typename T, void (*backend_fct_fmsub)(T, T, T, T*,void*)>
-        void call_backend_fct_fmsub(T* const a, T* const b, T* const c, T* d, void* ctx, int i)
+        void call_backend_fct_fmsub(T* const a, T* const b, T* const c, T* d, void* ctx, size_t i)
         {
             backend_fct_fmsub(a[i], b[i], -c[i], &d[i], ctx);
         }
 
         template<typename T, void (*backend_fct_fnmsub)(T, T, T, T*,void*)>
-        void call_backend_fct_fnmsub(T* const a, T* const b, T* const c, T* d, void* ctx, int i)
+        void call_backend_fct_fnmsub(T* const a, T* const b, T* const c, T* d, void* ctx, size_t i)
         {
             backend_fct_fnmsub(-a[i], b[i], -c[i], &d[i], ctx);
         }
@@ -178,7 +178,7 @@ namespace pene
                     {
                      if(iszeroing)
                      {
-                     for(int i=0;i< {{instruction.nb_elements}};i++)
+                     for(size_t i=0;i< {{instruction.nb_elements}};i++)
                      {
                       INS_InsertIfCall(ins, IPOINT_BEFORE, (AFUNPTR)msk_element_notset,
                       IARG_FAST_ANALYSIS_CALL,
@@ -206,7 +206,7 @@ namespace pene
                       }
                      }
                      else
-                      for(int i=0;i< {{instruction.nb_elements}};i++)
+                      for(size_t i=0;i< {{instruction.nb_elements}};i++)
                       {
                       INS_InsertIfCall(ins, IPOINT_BEFORE, (AFUNPTR)msk_element_set,
                       IARG_FAST_ANALYSIS_CALL,
@@ -271,7 +271,7 @@ namespace pene
                     {% elif instruction.isfma_avx512 == True %}
                     if(!ismasking || !REG_valid(mask_reg))
                      {
-                     for(int i=0;i< {{instruction.nb_elements}};i++)
+                     for(size_t i=0;i< {{instruction.nb_elements}};i++)
                      {%if instruction.optype != "fmaddsub" and instruction.optype != "fmsubadd" %}
                       {
                       INS_InsertCall(ins,IPOINT_BEFORE,(AFUNPTR){{which_fct_to_call(instruction.optype)}}<{{instruction.ins_precision.associated_type}}, OPERATION_IMPL::madd_{{instruction.ins_precision.associated_type}}>,
@@ -318,7 +318,7 @@ namespace pene
                     {
                      if(iszeroing)
                      {
-                     for(int i=0;i< {{instruction.nb_elements}};i++)
+                     for(size_t i=0;i< {{instruction.nb_elements}};i++)
                      {
                       INS_InsertIfCall(ins, IPOINT_BEFORE, (AFUNPTR)msk_element_notset,
                       IARG_FAST_ANALYSIS_CALL,
@@ -371,7 +371,7 @@ namespace pene
                       }
                      }
                      else
-                      for(int i=0;i< {{instruction.nb_elements}};i++)
+                      for(size_t i=0;i< {{instruction.nb_elements}};i++)
                       {
                       INS_InsertIfCall(ins, IPOINT_BEFORE, (AFUNPTR)msk_element_set,
                       IARG_FAST_ANALYSIS_CALL,

--- a/code_generator/templates/wrapper_template.txt
+++ b/code_generator/templates/wrapper_template.txt
@@ -59,11 +59,11 @@ namespace pene
         {
             dst[pos]=0;
         }
-        static bool PIN_FAST_ANALYSIS_CALL msk_element_set(IMULTI_ELEMENT_OPERAND* OpInfo, size_t pos)
+        static bool PIN_FAST_ANALYSIS_CALL msk_element_set(IMULTI_ELEMENT_OPERAND* OpInfo, UINT32 pos)
         {
           return OpInfo->ElementMaskValue(pos)==1;
         }
-        static bool PIN_FAST_ANALYSIS_CALL msk_element_notset(IMULTI_ELEMENT_OPERAND* OpInfo, size_t pos)
+        static bool PIN_FAST_ANALYSIS_CALL msk_element_notset(IMULTI_ELEMENT_OPERAND* OpInfo, UINT32 pos)
         {
           return OpInfo->ElementMaskValue(pos)==0;
         }
@@ -101,8 +101,8 @@ namespace pene
           auto xed = INS_XedDec(ins); // TODO check that pointer does not need to be freed
           auto iform = xed_decoded_inst_get_iform_enum(xed);
           {% if architecture_name == "avx512" or architecture_name == "fma" %}
-          bool ismasking=xed_decoded_inst_masking(xed);
-          bool iszeroing=xed_decoded_inst_zeroing(xed);
+          bool ismasking=(bool)xed_decoded_inst_masking(xed);
+          bool iszeroing=(bool)xed_decoded_inst_zeroing(xed);
           REG mask_reg = INS_MaskRegister(ins); 
           {% endif %}
 

--- a/include/replace/backend/double2float.h
+++ b/include/replace/backend/double2float.h
@@ -12,7 +12,7 @@ namespace pene
         static void sub_float(float a, float b, float* cptr, void*) noexcept { *cptr = a - b; }
         static void mul_float(float a, float b, float* cptr, void*) noexcept { *cptr = a * b; }
         static void div_float(float a, float b, float* cptr, void*) noexcept { *cptr = a / b; }
-        static void madd_float(float a, float b, float c, float* cptr, void*) noexcept { *cptr=fma(a,b,c); }
+        static void madd_float(float a, float b, float c, float* cptr, void*) noexcept { *cptr=fmaf(a,b,c); }
 
         static void add_double(double a, double b, double* cptr, void*) noexcept { *cptr = double(float(a) + float(b)); }
         static void sub_double(double a, double b, double* cptr, void*) noexcept { *cptr = double(float(a) * float(b)); }
@@ -22,7 +22,7 @@ namespace pene
           float a_f=float(a);
           float b_f=float(b);
           float c_f=float(c);
-          *cptr=double(fma(a_f,b_f,c_f));
+          *cptr=double(fmaf(a_f,b_f,c_f));
            }
         static void* init() noexcept { return nullptr; }
       };

--- a/include/replace/backend/invert_add_mul_impl.h
+++ b/include/replace/backend/invert_add_mul_impl.h
@@ -12,7 +12,7 @@ namespace pene
         static void sub_float(float a, float b, float* cptr, void*) noexcept { *cptr = a - b; }
         static void mul_float(float a, float b, float* cptr, void*) noexcept { *cptr = a + b; }// replace * with +
         static void div_float(float a, float b, float* cptr, void*) noexcept { *cptr = a / b; }
-        static void madd_float(float a, float b, float c, float* cptr, void*) noexcept { *cptr=fma(a,b,c); }
+        static void madd_float(float a, float b, float c, float* cptr, void*) noexcept { *cptr= fmaf(a,b,c); }
 
         static void add_double(double a, double b, double* cptr, void*) noexcept { *cptr = a + b; }
         static void sub_double(double a, double b, double* cptr, void*) noexcept { *cptr = a - b; }
@@ -29,13 +29,13 @@ namespace pene
         static void sub_float(float a, float b, float* cptr, void*) noexcept { *cptr = a - b; }
         static void mul_float(float a, float b, float* cptr, void*) noexcept { *cptr = a + b; }// replace * with +
         static void div_float(float a, float b, float* cptr, void*) noexcept { *cptr = a / b; }
-        static void madd_float(float a, float b, float c, float* cptr, void*) noexcept { *cptr=fma(a,b,c); }
+        static void madd_float(float a, float b, float c, float* cptr, void*) noexcept { *cptr= fmaf(a,b,c); }
 
         static void add_double(double a, double b, double* cptr, void*) noexcept { *cptr = a * b; }// replace + with *
         static void sub_double(double a, double b, double* cptr, void*) noexcept { *cptr = a - b; }
         static void mul_double(double a, double b, double* cptr, void*) noexcept { *cptr = a + b; }// replace * with +
         static void div_double(double a, double b, double* cptr, void*) noexcept { *cptr = a / b; }
-        static void madd_double(double a, double b, double c, double* cptr, void*) noexcept { *cptr=fma(a,b,c); }
+        static void madd_double(double a, double b, double c, double* cptr, void*) noexcept { *cptr= fma(a,b,c); }
         static void* init() noexcept { return nullptr; }
       };
 

--- a/include/replace/backend/test_backend.h
+++ b/include/replace/backend/test_backend.h
@@ -57,7 +57,7 @@ namespace pene
          
          static void madd_float(float a, float b, float c, float* cptr, void*) noexcept
           {
-           float temp =fma(a,b,c);
+           float temp = fmaf(a,b,c);
            auto  ans=bit_cast<uint32_t,float>(temp);
            ans &= 0xfffffff0;
            ans |= 0x9;

--- a/src/counters_module.cpp
+++ b/src/counters_module.cpp
@@ -25,7 +25,7 @@ namespace pene {
         }
         virtual void destroy(counters* local_counters) override
         {
-          for (int i = 0; i < counters::size; ++i)
+          for (size_t i = 0; i < counters::size; ++i)
           {
             c.array[i] += local_counters->array[i];
           }
@@ -130,7 +130,7 @@ namespace pene {
           PIN_WriteErrorMessage("ERROR: index<N)", 1000, PIN_ERR_FATAL, 0);
           PIN_ExitApplication(-1);
         }
-        if constexpr (N < counters::size)
+        if constexpr (N+1 < counters::size)
         {
           if (index > N)
           {

--- a/src/pene.cpp
+++ b/src/pene.cpp
@@ -25,7 +25,7 @@ INT32 Usage()
 
 
 
-VOID Fini(INT32 code, VOID* v)
+VOID Fini(INT32 code, VOID*)
 {
     // Write to a file since cout and cerr maybe closed by the application
     OutFile.setf(std::ios::showbase);

--- a/src/replace_module.cpp
+++ b/src/replace_module.cpp
@@ -230,9 +230,8 @@ namespace pene {
     case replace_module_internals::replace_modes::VERROU:
       {
       auto verrou_replace_instrmenter = new replace_module_internals::replace_inst_instrumenters<typename replace::backend::verrou>();
-      unsigned long seed=0;
-      srand(time(0));
-      seed=rand();
+      srand((uint)time(nullptr));
+      uint seed=(uint)rand();
       pene::replace::backend::verrou_set_seed (seed);
       if (rounding_mode == replace_module_internals::rounding_modes::NOTSPECIFIED)
         {

--- a/src/symbol_list_generator_module.cpp
+++ b/src/symbol_list_generator_module.cpp
@@ -35,7 +35,7 @@ namespace pene{
             update_counters(ins, c);
           }
 
-          for (int i = 0; i < counters::size && !has_fp_inst; ++i)
+          for (size_t i = 0; i < counters::size && !has_fp_inst; ++i)
           {
             has_fp_inst |= c.array[i] > 0;
           }
@@ -62,7 +62,7 @@ namespace pene{
         }
       }
 
-      for (int i = 0; i < counters::size && !has_fp_inst; ++i)
+      for (size_t i = 0; i < counters::size && !has_fp_inst; ++i)
       {
         has_fp_inst |= c.array[i] > 0;
       }
@@ -92,7 +92,7 @@ namespace pene{
   bool symbol_list_generator_module::validate()
   {
     std::cerr << "Checking configuration: symbol list generation - ";
-    auto filename = knob_exclist_gen.Value();
+    const auto& filename = knob_exclist_gen.Value();
 
     if (filename.empty())
     {
@@ -137,11 +137,11 @@ namespace pene{
       }
       PIN_AddFiniFunction([](INT32, void*) 
         {
-          for (auto img_name_it : sym_list)
+          for (auto& img_name_it : sym_list)
           {
-            auto img_name = img_name_it.first;
+            auto& img_name = img_name_it.first;
 
-            for (auto rtn_name : img_name_it.second)
+            for (auto& rtn_name : img_name_it.second)
             {
               sym_list_stream << std::left << std::setw(int(img_name_max_size) + 4) << img_name << rtn_name << "\n";
             }

